src/components/scenes/FioRequestConfirmationScene.js

```diff
@@ -0,0 +1,204 @@
+          continue
+        }
+      }
+
+      this.setState({
+        walletAddresses,
+        selectedFioAddress: walletAddresses[0].fioAddress
+      })
+    }
+  }
+
+  onNextPress = async () => {
+    const { walletAddresses, selectedFioAddress } = this.state
+    const walletAddress = walletAddresses.find(({ fioAddress }) => fioAddress === selectedFioAddress)
+
+    if (walletAddress) {
+      const { fioWallet } = walletAddress
+      const val = bns.div(this.props.amounts.nativeAmount, this.props.primaryCurrencyInfo.exchangeDenomination.multiplier, 18)
+      try {
+        if (!this.props.isConnected) {
+          showError(s.strings.fio_network_alert_text)
+          return
+        }
+        // checking fee
+        this.setState({ loading: true })
+        try {
+          const getFeeRes = await fioWallet.otherMethods.fioAction('getFee', { endPoint: 'new_funds_request', fioAddress: this.state.selectedFioAddress })
+          if (getFeeRes.fee) return showError(s.strings.fio_no_bundled_err_msg)
+        } catch (e) {
+          this.setState({ loading: false })
+          return showError(s.strings.fio_get_fee_err_msg)
+        }
+        // send fio request
+        await fioWallet.otherMethods.fioAction('requestFunds', {
+          payerFioAddress: this.props.fioModalData.fioAddress,
+          payeeFioAddress: this.state.selectedFioAddress,
+          payeeTokenPublicAddress: this.props.publicAddress,
+          amount: val,
+          tokenCode: this.props.primaryCurrencyInfo.exchangeCurrencyCode,
+          chainCode: this.props.chainCode || this.props.primaryCurrencyInfo.exchangeCurrencyCode,
+          memo: this.props.fioModalData.memo,
+          maxFee: 0
+        })
+        this.setState({ loading: false })
+        showToast(s.strings.fio_request_ok_body)
+        Actions.popTo(Constants.TRANSACTION_LIST)
+      } catch (error) {
+        this.setState({ loading: false })
+        showError(`${s.strings.fio_request_error_header}. ${error.json ? JSON.stringify(error.json.fields[0].error) : ''}`)
+      }
+    }
+  }
+
+  fiatAmount = (amount: string): string => {
+    const fiatPerCrypto = this.props.exchangeSecondaryToPrimaryRatio
+    return intl.formatNumber(fiatPerCrypto * parseFloat(amount), { toFixed: 2 }) || '0'
+  }
+
+  handleFioWalletChange = (something: string, index: number): void => {
+    this.setState({ selectedFioAddress: this.state.walletAddresses[index].fioAddress })
+  }
+
+  labelFromItem = (item: { fioAddress: string, label: string }): string => {
+    return item.label
+  }
+
+  render () {
+    const { loading, selectedFioAddress, walletAddresses } = this.state
+    const { primaryCurrencyInfo, secondaryCurrencyInfo, exchangeSecondaryToPrimaryRatio } = this.props
+    if (!primaryCurrencyInfo || !secondaryCurrencyInfo) return null
+    let finalValue, val
+    try {
+      finalValue = bns.div(this.props.amounts.nativeAmount, primaryCurrencyInfo.displayDenomination.multiplier, 18)
```

Comment:
Not clear variable name, use displayAmount.

Reply:
changed